### PR TITLE
AdSense/Doubleclick consent integration include consent parameter

### LINKED ISSUE #15381: AMP Ad component needs to send extra params to DFP for amp-consent/GDPR compliance
## What's the issue?

Regarding GDPR compliance, as provided by the amp-consent and amp-ad components.

After discussion with ade@google.com we have identified that for consistency and reporting purposes, DFP requires the following flags to be passed:

When consent is granted:

consent=true

When consent is not granted:

consent=false&npa=1

Doing network capture on ad requests shows that only npa=1 is being set, the consent param is not being set.

## How do we reproduce the issue?

If this is a bug please provide a public URL and ideally a reduced test case (e.g. on jsbin.com) that exhibits only your issue and nothing else.  Then provide step-by-step instructions for reproducing the issue:

1. Create a document with amp-ad components, pointing to DFP.
1. Add an amp-consent component.
1. Add the data-block-on-consent attribute to the amp-ad elements.

Expected: Filtering the network tab by 'pubads' should show requests with consent=true|false depending on the consent state
Actual: Filtering the network tab by 'pubads' DOES NOT show requests with consent=true|false depending on the consent state

## What browsers are affected?

All browsers

## Which AMP version is affected?

1525461683159

### PR DESCRIPTION
**Pull request by keithwrightbos**
Resolves #15381

### CODE REVIEW

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 1]
```diff
@@ -483,9 +483,26 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
    * @visibleForTesting
    */
   getPageParameters(consentState) {
+    let npa = null;
+    let consent = null;
+    switch (consentState) {
+      case undefined:
```
Comment (zhouyx) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: ditto

File: extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js [hunk 2]
```diff
@@ -290,14 +289,31 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
...
         ValidAdContainerTypes['AMP-FX-FLYING-CARPET']) ||
         enclosingContainers.includes(ValidAdContainerTypes['AMP-STICKY-AD']);
+    let npa = null;
+    let consent = null;
+    switch (consentState) {
+      case undefined:
+      case CONSENT_POLICY_STATE.UNKNOWN_NOT_REQUIRED:
+        break;
+      case CONSENT_POLICY_STATE.INSUFFICIENT:
+        consent = false;
```
Comment (bradfrizzell) [inline extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js]: I realize it's a line of duplication, but I think it is better readability to make each case explicit, and include npa='1' and a break in here as well.

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 3]
```diff
@@ -483,9 +483,26 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
...
+      case CONSENT_POLICY_STATE.UNKNOWN_NOT_REQUIRED:
+        break;
+      case CONSENT_POLICY_STATE.INSUFFICIENT:
+        consent = false;
```
Comment (bradfrizzell) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: same here

**Comment by rsimha** [comment]
@zhouyx To avoid the `jsdoc/require-param` error in line 155, add this before line 155...
```
/* eslint-disable jsdoc/require-param */
```
... and this after line 261:
```
/* eslint-enable jsdoc/require-param */
```

File: extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js [hunk 4]
```diff
@@ -152,6 +152,7 @@ let windowLocationQueryParameters;
  */
 let LayoutRectOrDimsDef;
 
+/* eslint-disable jsdoc/require-param */
```
Comment (rsimha) [inline extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js]: Or you could just fix the error with this line:
```
 * @param {function(!Array<AmpAdNetworkDoubleclickImpl>):?Object<string,string>} instances
```

**Comment by zhouyx** [comment]
Hey @keithwrightbos  @bradfrizzell we realize this needs to be cherry picked to canary to roll out next week. So we went ahead and merged this first. Please send following PRs to fix linter and address the comments from @bradfrizzell  : )

**Pull request merged**